homeassistant/components/stream/worker.py

```diff
@@ -101,29 +112,36 @@ def stream_worker(hass, stream, quit_event):
 
         # Reset segment on every keyframe
         if packet.is_keyframe:
-            # Calculate the segment duration by multiplying the presentation
-            # timestamp by the time base, which gets us total seconds.
-            # By then dividing by the sequence, we can calculate how long
-            # each segment is, assuming the stream starts from 0.
-            segment_duration = (packet.pts * packet.time_base) / sequence
```

Comment:
Not sure why but it seems like the original calculated the average segment length instead of the individual segment length.

Reply:
Was probably just me trying stuff until it worked some years back. Since keyframe's are generally at a set interval unless the feed has a dynamic frame rate the average and individual duration would probably be the same anyway. Good catch though.